library/core/src/iter/sources/successors.rs

```diff
@@ -1,11 +1,16 @@
 use crate::fmt;
 use crate::iter::FusedIterator;
 
-/// Creates a new iterator where each successive item is computed based on the preceding one.
+/// Creates an iterator which, starting from an initial item,
+/// computes each successive item from the preceding one.
 ///
-/// The iterator starts with the given first item (if any)
-/// and calls the given `FnMut(&T) -> Option<T>` closure to compute each item’s successor.
-/// The iterator will yield the `T`s returned from the closure.
+/// The iterator stores an optional item (`Option<T>`) and a successor closure (`impl FnMut(&T) -> Option<T>`).
+/// The iterator's `next` method returns the stored optional item and
+/// if it is `Some(val)` calls the stored closure on `&val` to compute and store its successor.
+/// The iterator will apply the closure successively to the stored option's value until the option is `None`.
+/// This also means that once the stored option is `None` it will remain `None`,
+/// as the closure will not be called again, so the created iterator is a `FusedIterator`.
+/// The iterator's items will be the initial item and all of its successors as calculated by the successor closure.
```

Comment:
This seems redundant to me – isn't this explained by the preceding sentences?

Reply:
This is mostly there as a direct result of the old text saying "The iterator will yield the `T`s returned from the closure.", and it does read pretty similar to the topline text ("Creates an iterator which, starting from an initial item, computes each successive item from the preceding one."), but includes slightly more detail (mentioning the successor closure which was explained in the previous paragraph.

I think it's still useful as a nice summary of the previous paragraph, which is quite detailed, and may be more readily understood than the detailed paragraph. But if you still think it is better without it, then I can remove it.

Reply:
Fair enough...